synapse/storage/databases/main/push_rule.py

```diff
@@ -859,8 +847,11 @@ async def copy_push_rules_from_room_to_room_for_user(
         user_push_rules = await self.get_push_rules_for_user(user_id)
 
         # Get rules relating to the old room and copy them to the new room
-        for rule in user_push_rules:
-            conditions = rule.get("conditions", [])
+        for rule, enabled in user_push_rules:
+            if not enabled:
+                continue
```

Comment:
I think it was probably a bug that we were copying over custom disabled rules on room upgrades? Though I think in practice custom push rules get deleted rather than disabled.